### ISSUE TITLE
[BUG] App hangs out or crushes after pasting picture in message

### ISSUE THREAD

**Issue by azkobain** [telegramdesktop/tdesktop#8509]
### Steps to reproduce
1. Open dialog with anybody
2. RB click on browser picture, choose "Copy"
3. RB click on dialog input, choose "Paste"
3***. Buttons combinations like CtrlC-CtrlV do nothing

### Expected behaviour
Tele shows dialog would I like prefer - send as file or as picture or smthng similiar
### Actual behaviour
Tele hangs out with other system or crushes

### Configuration
**Operating system:**
Debian 4.19.132-1 (2020-07-24)
**Version of Telegram Desktop:**
2.3.2
**Installation source (Linux Only)** - the official website / GitHub releases / flatpak / snap / distribution package:
The official website
**Used theme**: 
Ordinary

**Comment by gorghino** [comment]
It used to happen with previous Telegram Desktop versions: #7223 
This time it doesn't paste even clicking ctrl+v dozen of times.

**Comment by ilya-fedin** [comment]
I can't reproduce, have you disabled GTK integration?

**Comment by gorghino** [comment]
System side? No I haven't changed any GTK setting recently.
It used to work after GH-8038. I can generate debug log if you could explain how to trigger it.

**Comment by ilya-fedin** [comment]
> System side?

No, there are TDESKTOP_DISABLE_GTK_INTEGRATION environment variable and build flag that disables GTK integration and returns the issue. RPM Fusion maintainers defines it, for example.

You can check if GTK integration is enabled by running Telegram in terminal.

> I can generate debug log if you could explain how to trigger it.

I have no idea how to get debug logs for clipboard, it is on toolkit level.

**Comment by ilya-fedin** [comment]
Stop, if it is a crash, then where is crash report tag?

**Comment by gorghino** [comment]
TDESKTOP_DISABLE_GTK_INTEGRATION is not set in my env.
Running Telegram in terminal gives:

```
gorgo@Omen:~$ /home/gorgo/Telegram/Telegram -- %u
QApplication: invalid style override passed, ignoring it.
    Available styles: qt5ct-style, Windows, Fusion
Telegram-Message: 16:48:19.493: You can disable tray icon counter with TDESKTOP_DISABLE_TRAY_COUNTER and make it look better if it is monochrome.
error: : cannot open
error: : cannot open
error: : cannot open
```

> Stop, if it is a crash, then where is crash report tag?
Sorry, I forgot to mention that my TD doesn't hang or generate any crash log. CTRL+v doesn't do anything (unlike @azkobain apparently)

**Comment by ilya-fedin** [comment]
> Running Telegram in terminal gives

This is not good, please show `env | grep TDESKTOP`

**Comment by gorghino** [comment]
> > Running Telegram in terminal gives
> 
> This is not good, please show `env | grep TDESKTOP`

None

**Comment by ilya-fedin** [comment]
> None

That's impossible, there are should be another warning if you're using the latest version.

**Comment by ilya-fedin** [comment]
Stop, it's ok in case of static binary.

**Comment by ilya-fedin** [comment]
> Sorry, I forgot to mention that my TD doesn't hang or generate any crash log (unlike @azkobain apparently). CTRL+v doesn't do anything

I can't reproduce with GIMP :man_shrugging:

**Comment by gorghino** [comment]
> Stop, it's ok in case of static binary.

Yes I run it via Plank, but the command it uses is: /home/gorgo/Telegram/Telegram -- %u that is the static binary.
elementary OS 5.1.7 Hera
Linux 4.15.0-112-generic
GTK 3.22.30

I CAN copy/paste an image from GIMP, but not from Chrome (try https://www.facebook.com/photo/?fbid=10157010536736806&set=a.442424616805)

**Comment by ilya-fedin** [comment]
Oh, no way that I install that spyware software.

**Comment by gorghino** [comment]
Try this image: https://i.imgur.com/1ADl4cs.png

**Comment by ilya-fedin** [comment]
> I CAN copy/paste an image from GIMP

When there was the issue, it wasn't be possible to copy big images from GIMP (like 5000x5000), it was crashed with BadWindow error. Chrome doesn't crash since it handles the exception.

If you have problems only with Chrome, it is a different issue most likely and introduced by Chrome most likely.

**Comment by ilya-fedin** [comment]
> Try this image: https://i.imgur.com/1ADl4cs.png

No problems with pasting from Firefox nor GIMP.

**Comment by ilya-fedin** [comment]
btw, did you check Telegram log for GTK loading errors?

**Comment by gorghino** [comment]
Yes you're right. It seems to happen with Chrome only

**Comment by gorghino** [comment]
Plus, pasting the image on GIMP: stuck on loading.
Pasting the image on Skype: skype hangs.
Launching TD from terminal doesn't show any log related to clipboard nor GTK errors

**Comment by ilya-fedin** [comment]
> Launching TD from terminal doesn't show any log related to clipboard nor GTK errors

I'm talking about the log, not about the terminal output.

**Comment by gorghino** [comment]
Telegram Log log.txt ?
I don't see any error when I try to paste:

```
[2020.08.27 17:26:04] Launched version: 2003002, install beta: [FALSE], alpha: 0, debug mode: [FALSE]
[2020.08.27 17:26:04] Executable dir: /home/gorgo/Telegram/, name: Telegram
[2020.08.27 17:26:04] Initial working dir: /home/gorgo/
[2020.08.27 17:26:04] Working dir: /home/gorgo/.local/share/TelegramDesktop/
[2020.08.27 17:26:04] Command line: /home/gorgo/Telegram/Telegram -- %u -debug
[2020.08.27 17:26:04] Executable path before check: /home/gorgo/Telegram/Telegram
[2020.08.27 17:26:04] Logs started
[2020.08.27 17:26:04] Launcher filename: telegramdesktop.desktop
[2020.08.27 17:26:04] Checking for XDG Desktop Portal...
[2020.08.27 17:26:04] XDG Desktop Portal is present!
[2020.08.27 17:26:04] Getting DE, XDG_CURRENT_DESKTOP: 'Pantheon'
[2020.08.27 17:26:04] Getting DE, DESKTOP_SESSION: 'pantheon'
[2020.08.27 17:26:04] Getting DE, KDE_SESSION_VERSION: ''
[2020.08.27 17:26:04] Getting DE, GNOME_DESKTOP_SESSION_ID: 'this-is-deprecated'
[2020.08.27 17:26:04] DE: Gnome
[2020.08.27 17:26:04] Not using XDG Desktop Portal.
[2020.08.27 17:26:04] Connecting local socket to /run/user/1000/f3e26e130b1df48414208a7289de5404-{87A94AB0-E370-4cde-98D3-ACC110C5967D}...
[2020.08.27 17:26:04] This is the only instance of Telegram, starting server and app...
[2020.08.27 17:26:04] Moved logging from '/home/gorgo/.local/share/TelegramDesktop/log_start0.txt' to '/home/gorgo/.local/share/TelegramDesktop/log.txt'!
[2020.08.27 17:26:04] Primary screen DPI: 96
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAOpenSansRegular.ttf' loaded 'DAOpenSansRegular'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAVazirRegular.ttf' loaded 'DAVazirRegular'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAOpenSansRegularItalic.ttf' loaded 'DAOpenSansRegularItalic'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAVazirRegular.ttf' loaded 'DAVazirRegular'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAOpenSansSemiboldAsBold.ttf' loaded 'DAOpenSansSemibold'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAOpenSansSemiboldItalicAsBold.ttf' loaded 'DAOpenSansSemiboldItalic'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAOpenSansSemiboldAsBold.ttf' loaded 'DAOpenSansSemibold'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAOpenSansSemiboldItalicAsBold.ttf' loaded 'DAOpenSansSemiboldItalic'
[2020.08.27 17:26:04] Font: from ':/gui/fonts/DAVazirMediumAsBold.ttf' loaded 'DAVazirMedium'
[2020.08.27 17:26:04] App Info: reading settings...
[2020.08.27 17:26:04] App Info: reading encrypted settings...
[2020.08.27 17:26:04] Lang Info: Loaded cached, keys: 2723
[2020.08.27 17:26:04] OpenAL Logging Level: (not set)
[2020.08.27 17:26:04] Audio Playback Devices: Audio interno Stereo analogico;Audio interno Digital Stereo (HDMI)
[2020.08.27 17:26:04] Audio Playback Default Device: Audio interno Stereo analogico
[2020.08.27 17:26:04] Audio Capture Devices: Audio interno Stereo analogico;Monitor of Audio interno Digital Stereo (HDMI);Monitor of Audio interno Stereo analogico
[2020.08.27 17:26:04] Audio Capture Default Device: Audio interno Stereo analogico
[2020.08.27 17:26:04] System tray available: [FALSE]
[2020.08.27 17:26:04] Not using D-Bus global menu.
[2020.08.27 17:26:04] Using Unity launcher counter.
[2020.08.27 17:26:04] App Info: reading accounts info...
[2020.08.27 17:26:04] App Info: reading encrypted info...
[2020.08.27 17:26:04] App Info: reading map...
[2020.08.27 17:26:04] App Info: reading encrypted map...
[2020.08.27 17:26:04] App Info: reading encrypted user settings...
[2020.08.27 17:26:04] App Info: encrypted user settings read.
[2020.08.27 17:26:04] App Info: reading encrypted mtp data...
[2020.08.27 17:26:04] MTP Info: read keys, current: 5, to destroy: 0
[2020.08.27 17:26:04] Map read time: 2
[2020.08.27 17:26:04] App Info: reading encrypted mtp config...
[2020.08.27 17:26:04] TCP Error: network - Network unreachable
[2020.08.27 17:26:04] HTTP Error: network error 99 - Network unreachable
[2020.08.27 17:26:04] Export Info: Destroy top bar by controller removal.
[2020.08.27 17:27:33] Audio Info: recreating audio device and reattaching the tracks
[2020.08.27 17:27:35] Audio Info: Closing audio playback device.
[2020.08.27 17:28:15] Audio Info: recreating audio device and reattaching the tracks
[2020.08.27 17:28:17] Audio Info: Closing audio playback device.
[2020.08.27 17:28:44] TCP Error: network - Network unreachable
[2020.08.27 17:28:44] HTTP Error: network error 99 - Network unreachable
```

**Comment by ilya-fedin** [comment]
> I don't see any error when I try to paste

And shouldn't be, it is an app log, not a toolkit log

Everything is ok  guessing by the log.

**Comment by ilya-fedin** [comment]
@azkobain will you tell the crash report tag?

**Comment by azkobain** [comment]
Sorry, I'll not see my pc from work till the monday. So how can i get crash
report tag?

пт, 28 авг. 2020 г., 14:12 ilya-fedin <notifications@github.com>:

**Comment by ilya-fedin** [comment]
> So how can i get crash report tag?

You should get the crash report window on the next start of tdesktop after crash

**Comment by kav2k** [comment]
Possibly related: https://bugs.chromium.org/p/chromium/issues/detail?id=1114385

**Comment by kav2k** [comment]
Also, I can confirm that I'm experiencing the same issue (freezes) with Chrome -> Discord image copy-paste.

If possible, it would be nice to keep this issue open until Chrome deploys a fix (so that it's easier for people to find it), but it's not a problem with Telegram.

**Comment by Aokromes** [comment]
> Also, I can confirm that I'm experiencing the same issue (freezes) with Chrome -> Discord image copy-paste.
> 
> If possible, it would be nice to keep this issue open until Chrome deploys a fix (so that it's easier for people to find it), but it's not a problem with Telegram.

if you realize, i labeled 3rd party and don't closed it.

**Comment by kav2k** [comment]
With Chrome 85.0.4183.102 released yesterday, this should be fixed; I'm currently away from the machine where I can test it, can someone else confirm?

**Comment by gorghino** [comment]
Fixed and tested 85.0.4183.102 on Win10 and Linux
